crates/cli-support/src/js/mod.rs

```diff
@@ -322,12 +322,33 @@ impl<'a> Context<'a> {
         // Deno removed support for .wasm imports in https://github.com/denoland/deno/pull/5135
         // the issue for bringing it back is https://github.com/denoland/deno/issues/5609.
         format!(
-            "const file = new URL(import.meta.url).pathname;
-            const wasmFile = file.substring(0, file.lastIndexOf(Deno.build.os === 'windows' ? '\\\\' : '/') + 1) + '{}_bg.wasm';
-            const wasmModule = new WebAssembly.Module(Deno.readFileSync(wasmFile));
-            const wasmInstance = new WebAssembly.Instance(wasmModule, imports);
+            "const url = new URL(import.meta.url);
+
+            let wasmCode = '';
+            switch (url.protocol) {{
+                case 'file:':
+                    let file = new URL(import.meta.url).pathname;
+                    if (Deno.build.os === 'windows') {{
+                        if (file.startsWith('/')) file = file.substr(1);
+                        file = await Deno.realPath(file + '/../{module_name}_bg.wasm');
+                    }} else {{
+                        file = file.substring(0, file.lastIndexOf('/') + 1) + '{module_name}_bg.wasm';
+                    }}
+                    wasmCode = await Deno.readFile(file);
+                    break
+                case 'https:':
+                case 'http:':
+                    const wasm_url = import.meta.url.substring(0, import.meta.url.lastIndexOf('/') + 1) + '{module_name}_bg.wasm';
```

Comment:
It's more reliable to use real URL resolution instead of string manipulation. You can probably reuse this code: https://github.com/rustwasm/wasm-bindgen/blob/8594cae64b435cb5ab4d5233336fd15b3cb8cec2/crates/cli-support/src/js/mod.rs#L648

Reply:
Sorry and thank you.
I'm a newbie, I didn't know that URL has this kind of usage before this. So some codes used bad methods to try to get the `_bg.wasm` file in the same directory.
Now the code has become clean🎉